### DISCUSSION TITLE
Manager - jQuery should not be a modified script resource

### DISCUSSION THREAD

**Discussion by WebGeeksUnlimited** [PiranhaCMS/piranha.core#1888] [General]
I noticed that piranha-deps-dev.js or piranha-deps.min.js is basically jQuery with a bunch of modifications or additions.  This is bad when you want to use 3rd party components like Telerik kendo UI which rely on jQuery.  What ever changes were made are not compatible with jQuery behaving the same way as the stand alone js file.

I was adding a handful of back end pages and ran into the jQuery issue.

jQuery should be stock to maintain compatibility with other components that require jQuery. The other stuff should be in it's own js file.

If I comment the Piranha file mentioned & reference jQuery in the _Layout.cshtml, everything is fine, but you do get the Vue is not defined error in piranha.min.js.

-J

**Comment by tidyui** [comment]
The jQuery dist included in the Piranha manager should not be modified in any way, it is a dependency of Bootstrap which is used by the UI. jQuery is installed with npm:

* https://github.com/PiranhaCMS/piranha.core/blob/master/core/Piranha.Manager/package.json#L28

That package resource is then included (among other third party components) in the build script here:

* https://github.com/PiranhaCMS/piranha.core/blob/master/core/Piranha.Manager/gulpfile.js#L120

Could it be that you're relying on a newer version of jQuery than the one included in the manager?

Regards

**Reply by WebGeeksUnlimited** [reply]
OK, I understand how this was packaged in the Manager. Given that, the version used in Piranha (3.5.1), the Telerik Kendo UI came with 3.4.1, so I don't see that should be any issue. What I didn't get, is why it doesn't see jquery defined.

But then I noticed that the package doesn't load in the <head> section of the markup. kendo UI for some reason requires jQuery to be defined earlier than later.

So I scaffolded the Areas/Manager/Shared to included _Layout.cshtml & _ViewImports.cshtml to add the needed nuget packages and modify the layout slightly. I move the following from the current location to bottom of the <head> section.

    <environment include="Development">
        <script type="text/javascript" src="~/manager/assets/js/piranha-deps-dev.js?v=@version"></script>
    </environment>
    <environment exclude="Development">
        <script type="text/javascript" src="~/manager/assets/js/piranha-deps.min.js?v=@version"></script>
    </environment>

Once that was moved then kendo UI began to work.

One thing with adding script to the manager through the startup.cs, that always adds them to the bottom of the layout and no option to specify they should be placed in the <head> section.  There are odd scenarios where placement is important. But scaffolding the Areas is a way to work around that.

**Comment by tidyui** [comment]
You can actually specify if your custom script should be rendered before or after the core scripts from the manager UI, but maybe this is not enough for all use-cases.

https://github.com/PiranhaCMS/piranha.core/blob/master/core/Piranha.Manager/ManagerScriptDefinition.cs#L40

**Comment by WebGeeksUnlimited** [comment]
OK, I see thanks. I have managed to get the page fully working at this point.

On a slightly different area of the CMS with the front end menu (not the manager), is there a way to inject menu items that all call the same single page, but passes a parameter as part of the link?

For example, we have a product page, but we will display products based on a passed category parameter. In any case, I will need to figure out something else if not possible. It is my 1st time working with Piranha, so still wrapping my head around the architecture.
